src/garage/torch/algos/pearl.py

```diff
@@ -1,6 +1,635 @@
+                 recurrent=False,
+                 use_information_bottleneck=True,
+                 use_next_obs_in_context=False,
+                 meta_batch_size=64,
+                 num_steps_per_epoch=1000,
+                 num_initial_steps=100,
+                 num_tasks_sample=100,
+                 num_steps_prior=100,
+                 num_steps_posterior=0,
+                 num_extra_rl_steps_posterior=100,
+                 num_steps_per_eval=1000,
+                 batch_size=1024,
+                 embedding_batch_size=1024,
+                 embedding_mini_batch_size=1024,
+                 max_path_length=1000,
+                 discount=0.99,
+                 replay_buffer_size=1000000,
+                 reward_scale=1,
+                 update_post_train=1,
+                 eval_deterministic=True):
+
+        self.env = env
+        self._qf1 = qf
+        self._qf2 = copy.deepcopy(qf)
+        self._vf = vf
+        self._num_train_tasks = num_train_tasks
+        self._num_test_tasks = num_test_tasks
+        self._latent_dim = latent_dim
+
+        self._policy_mean_reg_coeff = policy_mean_reg_coeff
+        self._policy_std_reg_coeff = policy_std_reg_coeff
+        self._policy_pre_activation_coeff = policy_pre_activation_coeff
+        self._soft_target_tau = soft_target_tau
+        self._kl_lambda = kl_lambda
+        self._recurrent = recurrent
+        self._use_information_bottleneck = use_information_bottleneck
+        self._use_next_obs_in_context = use_next_obs_in_context
+
+        self._meta_batch_size = meta_batch_size
+        self._num_steps_per_epoch = num_steps_per_epoch
+        self._num_initial_steps = num_initial_steps
+        self._num_tasks_sample = num_tasks_sample
+        self._num_steps_prior = num_steps_prior
+        self._num_steps_posterior = num_steps_posterior
+        self._num_extra_rl_steps_posterior = num_extra_rl_steps_posterior
+        self._num_steps_per_eval = num_steps_per_eval
+        self._batch_size = batch_size
+        self._embedding_batch_size = embedding_batch_size
+        self._embedding_mini_batch_size = embedding_mini_batch_size
+        self.max_path_length = max_path_length
+        self._discount = discount
+        self._replay_buffer_size = replay_buffer_size
+        self._reward_scale = reward_scale
+        self._update_post_train = update_post_train
+        self._eval_deterministic = eval_deterministic
+        self.evaluator = None
+        self._task_idx = None
+
+        encoder_spec = self.get_env_spec(env[0](), latent_dim, 'encoder')
+        encoder_in_dim = int(np.prod(encoder_spec.input_space.shape))
+        encoder_out_dim = int(np.prod(encoder_spec.output_space.shape))
+        context_encoder = encoder_class(input_dim=encoder_in_dim,
+                                        output_dim=encoder_out_dim,
+                                        hidden_sizes=encoder_hidden_sizes)
+
+        self.policy = policy_class(
```

Comment:
is there a reason this can't be private?

Reply:
`LocalRunner` needs access to `policy`.

Reply:
can you implement this as a property instead? that way it is immutable

```python
def __init__(self, ...):
    self._policy = policy_class(...)

@property
def policy(self):
    """garage.torch.Policy: Most up-to-date policy."""
    return self._policy
```